### ISSUE TITLE
[Outdated] marktext-git

### ISSUE THREAD

**Issue by Technetium1** [chaotic-aur/packages#1570]
### If available, link to the latest build

[marktext-git.log](https://builds.garudalinux.org/repos/chaotic-aur/logs/marktext-git.log)

### Package name

`marktext-git`

### Latest build

`None`

### Latest version available

`0.17.0.r15.g2bb405a`

### Have you tested if the package builds in a clean chroot?

- [ ] Yes

### More information

Missing `electron11` since it was removed from community due to hitting end-of-life with v11.4.x on 2021-08-30. There's a patch in the comments for a fix using `electron17`, but nobody has confirmed if that patch results in a stable release yet.

**Comment by Technetium1** [comment]
I flagged the package out of date.

**Comment by Technetium1** [comment]
@alerque do you have any intention of switching to a currently supported version of Electron per [this patch](https://aur.archlinux.org/packages/marktext-git#comment-858677)?

**Comment by alerque** [comment]
Yes. I just tried getting Electron fixed on the stable package yesterday but got sidetracked. I'm on the road right now for a few days but will try to get a patch tested and applied as soon as I can.

**Comment by alerque** [comment]
That patch does *not* build for me. I don't want to push anything through until I can at least get it to build myself.

**Comment by Technetium1** [comment]
I definitely agree! Don't push until we know it's working in clean chroot :)

**Comment by alerque** [comment]
I tried a few more possible fixes and came up blank today. If anybody has working patches that build in a clean chroot and fix the electron version to be 17 as recommended upstream I would definitely give them a go and apply them if they check out.

**Comment by xiota** [comment]
This was dropped in d12c7e0bab684252009bc162da4516466047d261.  Will it be added back or permanently dropped?

[`aur/marktext-git`](https://aur.archlinux.org/packages/marktext-git) looks like updated to electron17?

**Comment by Technetium1** [comment]
@xiota That would do fine! As long as 17 is in extra.. Who knows for how long [since it's EOL](https://endoflife.date/electron).

**Comment by alerque** [comment]
Electron 17 will soon be dropped from [extra]. Both of the official Arch packages that depend on it have upstream releases supporting Electron 23 that we'll bump to as soon as we can spin up `elecron23` (I'm having issues with that). When it is dropped Marktext would have to depend on either users building it from the AUR or using an `electron17-bin` package.

Please also do open an issue upstream on Marktext because it is ridiculous that their current stable release uses an EOL version of Electron.

**Comment by xiota** [comment]
````
$ pacman -Ss electron
extra/electron 22.3.3-1
    Build cross platform desktop apps with web technologies
extra/electron17 17.4.11-6
    Build cross platform desktop apps with web technologies
extra/electron19 19.1.9-5
    Build cross platform desktop apps with web technologies
extra/electron20 20.3.12-2
    Build cross platform desktop apps with web technologies
extra/electron21 21.4.4-1
    Build cross platform desktop apps with web technologies
extra/electron24 24.4.0-1
    Build cross platform desktop apps with web technologies
````

I'm predisposed to the idea of banning electron apps because they break too easily.  Potential exceptions for some of them.

**Comment by Technetium1** [comment]
The nature of a browser is to do weird things sometimes. Not a valid reason for banning them. Size is often a problem, but not enough to make assumptions. It *is* valid to not compile end of life versions of Electron though!

**Comment by alerque** [comment]
I just poked around upstream and it looks like Marktext development is pretty much abandoned, and has been for a year or so.

**Comment by xiota** [comment]
The homepage is held by a cybersquatter.
